server/etcdserver/api/v3rpc/maintenance.go

```diff
@@ -89,10 +95,10 @@ func NewMaintenanceServer(s *etcdserver.EtcdServer, hn HealthNotifier) pb.Mainte
 
 func (ms *maintenanceServer) Defragment(ctx context.Context, sr *pb.DefragmentRequest) (*pb.DefragmentResponse, error) {
 	ms.lg.Info("starting defragment")
-	if ms.stopServingOnDefrag {
-		ms.hn.StopServe("defrag is active")
-		defer ms.hn.StartServe()
-	}
+	ms.defragC <- defragStarted
+	defer func() {
+		ms.defragC <- defragFinished
+	}()
```

Comment:
Don't think that maintenanceServer needs to be aware that signal is sent via channels.
```suggestion
	ms.notify.DefragStarted()
	defer ms.notify.DefragFinished()
```

Reply:
Done